net/net.go

```diff
@@ -154,3 +156,47 @@ func ParseIPCIDRs(cidrs []string) (*netipx.IPSet, error) {
 
 	return ips, nil
 }
+
+// SchemeHost parses URI string (without #fragment part) and returns schema used in this URI as first return value and
+// host:port part as second return value. Port is never omitted, if no port is specified in URI, default port for given
+// schema is used. If URI is invalid or port could not be detected, error is returned.
+func SchemeHost(input string) (string, string, error) {
+	u, err := url.ParseRequestURI(input)
+	if err != nil {
+		return "", "", err
+	}
+
+	// endpoint address cannot contain path, the rest is not case sensitive
+	s, h := strings.ToLower(u.Scheme), strings.ToLower(u.Host)
+	if h == "" {
```

Comment:
Please move empty check before ToLower.
Also looks like scheme can be empty?

Reply:
If scheme is empty, the https://pkg.go.dev/net/url#ParseRequestURI above will fail, and this line of code won't be executed:

```
$ cat main.go 
package main

import (
	"fmt"
	"net/url"
)

func main() {
	u, err := url.ParseRequestURI("example.com")
	fmt.Printf("u: %v, err: %v\n", u, err)
}
$ go build main.go 
$ ./main 
u: <nil>, err: parse "example.com": invalid URI for request
```

Reply:
However, I will add check for empty scheme as well. Just to be on the safe side.